Clearup mod AIs.

GraionDilach opened the pull request:
The percentages might sound lame but having the summary of the riflemen and rocket infantry above 100 will make sure no techlevel gets stuck - excluding the RA Naval AI, which doesn't even use infantry at all (notsure if it should be allowed to spam those two).

Fixes #11797.

Commit:
Update Red Alert AI.

Increase land AI infantry percentages.
Add medic and mechanic to Normal and Turtle AI.
Limit Engineers at Rush AI.

Commit:
Update Tiberian Dawn AI.

Increase all AI infantry percentages.
Limit Engineers at Rush AI.

Commit:
Update Dune 2000 AI.

Increase all AI combat infantry percentages.
Remove nonbuildable and deprecated infantry from AI UnitsToBuild list.

GraionDilach commented:
A question of mine - is there a reason why the AI doesn't build commandos in TD/RA? Woulda add it as part of this PR if there isn't.

obrakmann commented:
Probably the same reason why other "special" actors aren't listed: the AI doesn't know how to work with them. For Tanya and the Commando you could argue that they could work at least half-way. I don't think it would enhance the gameplay much, if any, though.

obrakmann commented on mods/d2k/rules/ai.yaml:
```diff
@@ -48,14 +48,12 @@ Player:
-			saboteur: 0.5%
+			light_inf: 65%
+			trooper: 40%
+			thumper: 1%
```
I'd leave those out, the AI doesn't know what to do with them (ie., how and where to deploy them).

GraionDilach commented:
My main reason is that in RA the Allied AI only use the two basic infantry tbh. Guess I should remove thumpers as well then, since the AI cannot deploy those..

GraionDilach commented:
Dropped the medic-to-thumper rename commit and removed the now-defunct medic entries in the last commit.

obrakmann commented:
Well, I think medics and mechanics *could* work for the AI, since they just use a weapon now. It might be worth testing.

GraionDilach commented:
After a test, it appears #8768 and related made the AI enough effective at handling medics/mechanics to add them. (They always used a weapon.) Added them to Normal and Turtle AI.

Also unitlimited the engineerspam of the Rush AIs.

Rebased and squashed the commits to one per mod.

Commit:
Update Tiberian Sun AI.

Increase infantry percentages.
Limit the AI repair units.

GraionDilach commented:
Forgot to update TS. >_>

WolfGaming commented:
This looks good, will test it later tonight.

WolfGaming commented:
This works, the ai no longer gets bored though they are still shitty so :+1: from me.

obrakmann commented:
Looks ok, :+1:

obrakmann commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/ca96c360473ec56c1e4557a68220ceeee6f02b02)
